cmd/gossamer/gossamer_test.go

```diff
@@ -0,0 +1,154 @@
+package main_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/build"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	cfg "github.com/ChainSafe/gossamer/config"
+	"github.com/ChainSafe/gossamer/p2p"
+	"github.com/ChainSafe/gossamer/polkadb"
+	"github.com/inconshreveable/log15"
+	"github.com/rendon/testcli"
+)
+
+var binaryname = "gossamer-test"
+
+const configTest = "config-test.toml"
+const timeFormat = "2006-01-02T15:04:05-0700"
+
+func setup() {
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
+		gopath = build.Default.GOPATH
+	}
+	r := exec.Command("go", "build", "-o", gopath+"/bin/gossamer-test")
+	err := r.Run()
+	if err != nil {
+		log15.Crit("could not execute binary", "executable", binaryname, "err", err)
+		os.Exit(1)
+	}
+	run := exec.Command(`gossamer-test`)
+	err = run.Run()
+	if err != nil {
+		log15.Crit("could not execute binary", "executable", binaryname, "err", err)
+		os.Exit(1)
+	}
+	fp, err := os.Create(configTest)
+	if err != nil {
+		log15.Crit("could not create test config", "config", configTest, "err", err)
+		os.Exit(1)
+	}
+
+	testConfig := fmt.Sprintf("%s%s%s%v%s%s%v%s%s%s",
+		"[ServiceConfig]\n", "BootstrapNodes = []\n", "Port = ",
+		7005, "\n", "RandSeed = ", 0, "\n\n", "[DbConfig]\n",
+		"Datadir = \"\"\x0A")
+
+	_, err = fp.WriteString(testConfig)
+	if err != nil {
+		log15.Crit("could not write to test config", "config", "config-test.toml", "err", err)
+		os.Exit(1)
+	}
+}
+
+func teardown() {
+	err := os.Chdir("../gossamer")
+	if err != nil {
+		log15.Error("could not change dir", "err", err)
+		os.Exit(1)
+	}
+	if err := os.RemoveAll("./chaindata"); err != nil {
+		log15.Warn("removal of temp directory bin failed", "err", err)
+	}
+	if err := os.RemoveAll("./config-test.toml"); err != nil {
+		log15.Warn("removal of temp config.toml failed", "err", err)
+	}
+}
+
+func TestInitialOutput(t *testing.T) {
+	setup()
+	testcli.Run("gossamer-test")
+	if !testcli.Success() {
+		teardown()
+		t.Fatalf("Expected to succeed, but failed: %s", testcli.Error())
+	}
+	output := fmt.Sprintf("%s%v%s", "t=", time.Now().Format(timeFormat), " lvl=info msg=\"🕸️ starting p2p service\" blockchain=gossamer\x0A")
+	if !testcli.StdoutContains(output) {
+		teardown()
+		t.Fatalf("Expected %q to contain %q", testcli.Stdout(), output)
+	}
+	if !reflect.DeepEqual(testcli.Stdout(), output) {
+		teardown()
+		t.Fatalf("actual = %s, expected = %s", testcli.Stdout(), output)
+	}
+	defer teardown()
```

Comment:
nit picky stuff -- I think you can defer the teardown on line 83 and it will get called when `t.Fatalf` gets called.